pkg/cmd/server/origin/legacy.go

```diff
@@ -186,9 +189,38 @@ func LegacyStorage(storage map[schema.GroupVersion]map[string]rest.Storage) map[
 	for _, gvStorage := range storage {
 		for resource, s := range gvStorage {
 			if OriginLegacyResources.Has(resource) || OriginLegacySubresources.Has(resource) {
-				legacyStorage[resource] = s
+				// We want *some* our legacy resources to orphan by default instead of garbage collecting.
+				// Kube only did this for a select few resources which were controller managed and established links
+				// via a workload controller.  In openshift, these will all conform to registry.Store so we
+				// can actually wrap the "normal" storage here.
+				switch resource {
+				case "buildConfigs":
```

Comment:
> Surprised we still mix case these

Yeah, its even weirder in that appear to be lowercased when we expose them.  Prexisting debt I don't plan to fix here.